### PR TITLE
[DOC] Rename OTX to OpenVINO™ Training Extensions

### PR DESCRIPTION
**Pull request by supersoob**
Please comment if you find any OTX or something wrong

### CODE REVIEW

File: tests/test_suite/ARCHITECTURE.md [hunk 1]
```diff
@@ -883,13 +883,13 @@ Let's consider all the methods of the abstract test parameters interface class o
 
 ```python
 @abstractmethod
-def test_case_class(self) -> Type[OTXTestCaseInterface]:
+def test_case_class(self) -> Type[OpenVINO™ Training ExtensionsTestCaseInterface]:
```
Comment (sungmanc) [inline tests/test_suite/ARCHITECTURE.md]:
```suggestion
def test_case_class(self) -> Type[OTXTestCaseInterface]:
```

File: tests/test_suite/ARCHITECTURE.md [hunk 2]
```diff
@@ -843,7 +843,7 @@ a whole:
 The constructor of the class `OTXTestHelper` has the following declaration
 
 ```python
-def __init__(self, test_creation_parameters: OTXTestCreationParametersInterface):
+def __init__(self, test_creation_parameters: OpenVINO™ Training ExtensionsTestCreationParametersInterface):
```
Comment (sungmanc) [inline tests/test_suite/ARCHITECTURE.md]:
```suggestion
def __init__(self, test_creation_parameters: OTXTestCreationParametersInterface):
```

File: tests/test_suite/ARCHITECTURE.md [hunk 3]
```diff
@@ -856,7 +856,7 @@ most of information required to connect the test suite with a concrete algo back
 All the methods of the interface class are abstract methods without parameters that return
 structures making this connection.
 
-Example of such implementation is the class `DefaultOTXTestCreationParametersInterface` that
+Example of such implementation is the class `DefaultOpenVINO™ Training ExtensionsTestCreationParametersInterface` that
```
Comment (sungmanc) [inline tests/test_suite/ARCHITECTURE.md]:
```suggestion
Example of such implementation is the class `DefaultOTXTestCreationParametersInterface` that
```

File: tests/test_suite/ARCHITECTURE.md [hunk 4]
```diff
@@ -711,7 +711,7 @@ The constructor works as follows:
   - call constructor of the current action as
     `cur_action = action_cls(**cur_params)`
   - wraps the current action with the class `OTXTestStage` as follows:
-    `cur_stage = OTXTestStage(action=cur_action, stages_storage=self)`
+    `cur_stage = OpenVINO™ Training ExtensionsTestStage(action=cur_action, stages_storage=self)`
```
Comment (sungmanc) [inline tests/test_suite/ARCHITECTURE.md]:
```suggestion
    `cur_stage = OTXTestStage(action=cur_action, stages_storage=self)`
```

File: tests/test_suite/ARCHITECTURE.md [hunk 5]
```diff
@@ -602,7 +602,7 @@ the function has the declaration
 
 ```python
 def generate_otx_integration_test_case_class(
-    test_actions_classes: List[Type[BaseOTXTestAction]],
+    test_actions_classes: List[Type[BaseOpenVINO™ Training ExtensionsTestAction]],
```
Comment (sungmanc) [inline tests/test_suite/ARCHITECTURE.md]:
```suggestion
    test_actions_classes: List[Type[BaseOTXTestAction]],
```

File: tests/test_suite/ARCHITECTURE.md [hunk 6]
```diff
@@ -253,23 +253,23 @@ action. For each instance of a test action an instance of the class `OTXTestStag
...
+  (all the test case classes are derived from OpenVINO™ Training ExtensionsTestStagesStorageInterface)
 
 The `stages_storage` instance is also kept inside `OTXTestStage`, it will be used to get for each
 stage its dependencies.
 Note that the abstract interface class `OTXTestStagesStorageInterface` has the only abstract method
 `get_stage` with declaration
 
 ```python
-def get_stage(self, name: str) -> "OTXTestStage":
+def get_stage(self, name: str) -> "OpenVINO™ Training ExtensionsTestStage":
```
Comment (sungmanc) [inline tests/test_suite/ARCHITECTURE.md]:
```suggestion
def get_stage(self, name: str) -> "OTXTestStage":
```

File: tests/test_suite/ARCHITECTURE.md [hunk 7]
```diff
@@ -253,23 +253,23 @@ action. For each instance of a test action an instance of the class `OTXTestStag
...
+def __init__(self, action: BaseOpenVINO™ Training ExtensionsTestAction, stages_storage: OpenVINO™ Training ExtensionsTestStagesStorageInterface):
 ```
 
 - The `action` parameter here is the instance of action that is wrapped.
   It is kept inside the `OTXTestStage` instance.
 - The `stages_storage` here is an instance of a class that allows to get a stage by name, this will
   be a test case class that connects all the test stages between each other and keeps in its fields
   results of all test stages between tests
-  (all the test case classes are derived from OTXTestStagesStorageInterface)
+  (all the test case classes are derived from OpenVINO™ Training ExtensionsTestStagesStorageInterface)
```
Comment (sungmanc) [inline tests/test_suite/ARCHITECTURE.md]:
```suggestion
  (all the test case classes are derived from OTXTestStagesStorageInterface)
```

File: tests/test_suite/ARCHITECTURE.md [hunk 8]
```diff
@@ -1258,7 +1258,7 @@ As an example of the test parameters class see
   `external/deep-object-reid/tests/test_otx_training.py`
   -- the latter is more interesting, since deep-object-reid algo backend is different w.r.t. the
   mmdetection algo backend, and we implemented the default test case parameter class
-  `DefaultOTXTestCreationParametersInterface` mostly for mmdetection.
+  `DefaultOpenVINO™ Training ExtensionsTestCreationParametersInterface` mostly for mmdetection.
```
Comment (sungmanc) [inline tests/test_suite/ARCHITECTURE.md]:
```suggestion
  `DefaultOTXTestCreationParametersInterface` mostly for mmdetection.
```

File: tests/test_suite/ARCHITECTURE.md [hunk 9]
```diff
@@ -1281,7 +1281,7 @@ The test case class should be implemented as follows:
 
 - The test class should have a static field `helper` defined as follows:
   ```python
-  helper = OTXTestHelper(<test parameters class>())
+  helper = OpenVINO™ Training ExtensionsTestHelper(<test parameters class>())
```
Comment (sungmanc) [inline tests/test_suite/ARCHITECTURE.md]:
```suggestion
  helper = OTXTestHelper(<test parameters class>())
```

**Review by sungmanc — CHANGES_REQUESTED** [review]

File: README.md [hunk 10]
```diff
@@ -66,7 +66,7 @@ Moving forward, OTX provides the following features:
 ### v1.0.0 (1Q23)
 
 - Installation through PyPI
-  - Package will be renamed as OTX (OpenVINO Training eXtensions)
+  - Package will be renamed as OpenVINO™ Training Extensions (OpenVINO Training eXtensions)
```
Comment (eunwoosh) [inline README.md]:
```suggestion
  - Package will be renamed as OpenVINO™ Training Extensions
```
Isn't it needed?

**Review by eunwoosh — DISMISSED** [review]

File: QUICK_START_GUIDE.md [hunk 11]
```diff
@@ -259,7 +259,7 @@ Example of the command line to start object detection training:
 
 ```
 
-It is also possible to start training by omitting the template and just passing the paths to dataset roots, then the [auto-configuration](https://openvinotoolkit.github.io/training_extensions/explanation/additional_features/auto_configuration.html) will be enabled. Based on the dataset OTX will choose the task type and template with the best accuracy/speed trade-off.
+It is also possible to start training by omitting the template and just passing the paths to dataset roots, then the [auto-configuration](https://openvinotoolkit.github.io/training_extensions/explanation/additional_features/auto_configuration.html) will be enabled. Based on the dataset OpenVINO™ Training Extensions will choose the task type and template with the best accuracy/speed trade-off.
```
Comment (supersoob) [inline QUICK_START_GUIDE.md]:
```suggestion
It is also possible to start training by omitting the template and just passing the paths to dataset roots, then the [auto-configuration](https://openvinotoolkit.github.io/training_extensions/explanation/additional_features/auto_configuration.html) will be enabled. Based on the dataset, OpenVINO™ Training Extensions will choose the task type and template with the best accuracy/speed trade-off.
```

File: tests/test_suite/ARCHITECTURE.md [hunk 12]
```diff
@@ -116,7 +116,7 @@ algo backends.
 
 Note that each test action class MUST have the following properties:
 
-- it MUST be derived from the base class `BaseOTXTestAction`;
+- it MUST be derived from the base class `BaseOpenVINO™ Training ExtensionsTestAction`;
```
Comment (supersoob) [inline tests/test_suite/ARCHITECTURE.md]:
```suggestion
- it MUST be derived from the base class `BaseOTXTestAction`;
```

File: tests/test_suite/ARCHITECTURE.md [hunk 13]
```diff
@@ -253,7 +253,7 @@ action. For each instance of a test action an instance of the class `OTXTestStag
 It's constructor has declaration
 
 ```python
-def __init__(self, action: BaseOTXTestAction, stages_storage: OTXTestStagesStorageInterface):
+def __init__(self, action: BaseOpenVINO™ Training ExtensionsTestAction, stages_storage: OpenVINO™ Training ExtensionsTestStagesStorageInterface):
```
Comment (supersoob) [inline tests/test_suite/ARCHITECTURE.md]:
```suggestion
def __init__(self, action: BaseOTXTestAction, stages_storage: OTXTestStagesStorageInterface):
```

**Review by sungmanc — DISMISSED** [review]
Thanks for the quick fix

File: README.md [hunk 14]
```diff
@@ -66,7 +66,7 @@ Moving forward, OTX provides the following features:
 ### v1.0.0 (1Q23)
 
 - Installation through PyPI
-  - Package will be renamed as OTX (OpenVINO Training eXtensions)
+  - Package will be renamed as OpenVINO™ Training Extensions
```
Comment (kprokofi) [inline README.md]: It is actually now no new information since it is already OpenVINO™ Training Extensions.

**Review by kprokofi — COMMENTED** [review]
Will you change OTX in .rst docs as well?

File: QUICK_START_GUIDE.md [hunk 15]
```diff
@@ -77,7 +77,7 @@ source venv/otx/bin/activate
 pytest tests/unit
 ```
 
-## OTX CLI commands
+## OpenVINO™ Training Extensions CLI commands
 
 Below, all possible otx CLI commands are presented with some general examples of how to run specific functionality. We also have [dedicated tutorials](https://openvinotoolkit.github.io/training_extensions/tutorials/index.html) in our documentation with life-practical examples on specific datasets for each task.
```
Comment (supersoob) [inline QUICK_START_GUIDE.md]:
```suggestion
Below, all possible CLI commands are presented with some general examples of how to run specific functionality. We also have [dedicated tutorials](https://openvinotoolkit.github.io/training_extensions/tutorials/index.html) in our documentation with life-practical examples on specific datasets for each task.
```

File: QUICK_START_GUIDE.md [hunk 16]
```diff
@@ -191,7 +191,7 @@ coco_data_root
 
 Then pass the path to `coco_data_root` to both root options: `--train-data-roots coco_data_root --val-data-roots coco_data_root`. However, if you store your training set and validation separately - provide paths to both accordingly.
 
-OTX supports also auto-split functionality. If you don't have a prepared validation set - the Datumaro manager will run a random auto-split and will save the final dataset to `splitted_dataset` folder inside the otx workspace folder. This split can be further used for training.
+OpenVINO™ Training Extensions supports also auto-split functionality. If you don't have a prepared validation set - the Datumaro manager will run a random auto-split and will save the final dataset to `splitted_dataset` folder inside the otx workspace folder. This split can be further used for training.
```
Comment (supersoob) [inline QUICK_START_GUIDE.md]:
```suggestion
OpenVINO™ Training Extensions supports also auto-split functionality. If you don't have a prepared validation set - the Datumaro manager will run a random auto-split and will save the final dataset to `splitted_dataset` folder inside the workspace folder. This split can be further used for training.
```

File: README.md (hunk 14 above)
Comment (supersoob) [reply README.md]: We've already change OTX in .rst. I have no idea about this question. @chuneuny-emily Could you answer?

Comment (chuneuny-emily) [reply README.md]: Let's update it like,
- Package Installation via PyPI
  - OpenVINO™ Training Extensions installation will be supported via PyPI.

**Comment by supersoob** [comment]
@chuneuny-emily @sungmanc @chuneuny-emily I think I checked all. Could you approve?

**Review by chuneuny-emily — APPROVED** [review]

**Review by goodsong81 — APPROVED** [review]
Thank you for the fix!
